src/buffer/base.ts

```diff
@@ -0,0 +1,447 @@
+   */
+  stringColumn(name: string, value: string): SenderBuffer {
+    this.writeColumn(
+      name,
+      value,
+      () => {
+        this.checkCapacity([value], 2 + value.length);
+        this.write('"');
+        this.writeEscaped(value, true);
+        this.write('"');
+      },
+      "string",
+    );
+    return this;
+  }
+
+  /**
+   * Write a boolean column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {boolean} value - Column value, accepts only boolean values.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  booleanColumn(name: string, value: boolean): SenderBuffer {
+    this.writeColumn(
+      name,
+      value,
+      () => {
+        this.checkCapacity([], 1);
+        this.write(value ? "t" : "f");
+      },
+      "boolean",
+    );
+    return this;
+  }
+
+  /**
+   * Write a float column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {number} value - Column value, accepts only number values.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  abstract floatColumn(name: string, value: number): SenderBuffer;
+
+  /**
+   * Write an integer column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {number} value - Column value, accepts only number values.
+   * @return {Sender} Returns with a reference to this sender.
+   * @throws Error if the value is not an integer
+   */
+  intColumn(name: string, value: number): SenderBuffer {
+    if (!Number.isInteger(value)) {
+      throw new Error(`Value must be an integer, received ${value}`);
+    }
+    this.writeColumn(name, value, () => {
+      const valueStr = value.toString();
+      this.checkCapacity([valueStr], 1);
+      this.write(valueStr);
+      this.write("i");
+    });
+    return this;
+  }
+
+  /**
+   * Write a timestamp column with its value into the buffer.
+   *
+   * @param {string} name - Column name.
+   * @param {number | bigint} value - Epoch timestamp, accepts numbers or BigInts.
+   * @param {string} [unit=us] - Timestamp unit. Supported values: 'ns' - nanoseconds, 'us' - microseconds, 'ms' - milliseconds. Defaults to 'us'.
+   * @return {Sender} Returns with a reference to this sender.
+   */
+  timestampColumn(
+    name: string,
+    value: number | bigint,
+    unit: TimestampUnit = "us",
+  ): SenderBuffer {
+    if (typeof value !== "bigint" && !Number.isInteger(value)) {
+      throw new Error(`Value must be an integer or BigInt, received ${value}`);
+    }
+    this.writeColumn(name, value, () => {
+      const valueMicros = timestampToMicros(BigInt(value), unit);
+      const valueStr = valueMicros.toString();
+      this.checkCapacity([valueStr], 1);
+      this.write(valueStr);
+      this.write("t");
+    });
+    return this;
+  }
+
+  /**
+   * Closing the row after writing the designated timestamp into the buffer.
```

Comment:
nit: the tsdoc should start with a verb, e.g. "Closes the row..." or "Finalizes the row...". Also applies to other tsdocs that start with "Closing...".

Reply:
fixed in https://github.com/questdb/nodejs-questdb-client/pull/49/commits/4f01717b39768e3ee1be6b1332186206dcddd4cf